cmd/clusterctl/pkg/client/cluster/inventory.go

```diff
@@ -200,53 +203,35 @@ func (p *inventoryClient) Create(m clusterctlv1.Provider) error {
 }
 
 func (p *inventoryClient) List() ([]clusterctlv1.Provider, error) {
-	return p.list(listOptions{})
-}
-
-type listOptions struct {
-	Namespace string
-	Name      string
-	Type      clusterctlv1.ProviderType
+	providerList, err := p.list()
```

Comment:
Nit: This function (`func (p *inventoryClient) List()`) isn't doing much any more, is it still necessary? It doesn't do any additional error handling is simply returning `.Items` from the provider list. Perhaps this could be removed or lower-case `list()`'s implemenation could live in here.

Reply:
This makes sense, 
However, if I change `List() ([]clusterctlv1.Provider, error)` --> `List() (*clusterctlv1.ProviderList, error)` I have to change bits of code unrelated to this PR, so I send a separated cleanup PR